File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/DTOS/CartDTOS/CartItemDTO.java
package app.aurafitbackend.DTOS.CartDTOS;

import app.aurafitbackend.DTOS.DisplayDTOS.ProductVariantDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class CartItemDTO {

    private Long id;
    private ProductVariantDTO variant;
    private Integer quantity;
    private BigDecimal unitPrice;
    private Long cartId;

}

File: aurafit-frontend/src/Models/DTOS/CartItemDTO.ts
import {ProductVariant} from "../ProductVariant.ts";
import {ProductVariantDTO} from "./ProductVariantDTO.ts";


export class CartItemDTO {
    id: number;
    variant: ProductVariantDTO;
    quantity: number;
    unitPrice: number;
    cartId: number;

    constructor(id: number, variant: ProductVariantDTO, quantity: number, unitPrice: number, cartId: number) {
        this.id = id;
        this.variant = variant;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
        this.cartId = cartId;
    }
}